spec/lib/artifact-search.spec.js

```diff
@@ -0,0 +1,125 @@
+            });
+        });
+        it("returns an array of URLs", () => {
+            responseHandlerMock.resolveLinkHeaders.andReturn([
+                "http://www.example.com/refName/artifact/exampleData1",
+                "http://www.example.com/refName/artifact/exampleData2"
+            ]);
+            responseHandlerMock.resolveLink.andCallFake((options, val) => {
+                return val;
+            });
+
+            return instance.search().then((urls) => {
+                expect(urls).toEqual(jasmine.any(Array));
+            });
+        });
+        it("calls responseHandler.handleErrorResponse on error", () => {
+            requestPromiseMock.post.andReturn(bluebird.reject());
+
+            return instance.search().then(() => {
+                expect(responseHandlerMock.handleErrorResponse).toHaveBeenCalled();
+            });
+        });
+        describe(".validateSearchItem()", () => {
+            var searchParameters;
+
+            beforeEach(() => {
+                spyOn(bluebird, "reject").andCallThrough();
+                spyOn(bluebird, "resolve").andCallThrough();
+                responseHandlerMock.resolveLinkHeaders.andReturn([]);
+            });
+            it("rejects if the query violates the regex", () => {
+                searchParameters = {
+                    search: "\\=Invalid Query"
+                };
+
+                return instance.search(searchParameters).then(jasmine.fail, () => {
+                    expect(bluebird.reject).toHaveBeenCalled();
+                });
+            });
+            it("resolves with a valid query", () => {
+                searchParameters = {
+                    search: "Valid=Query"
+                };
+
+                return instance.search(searchParameters).then(() => {
+                    expect(bluebird.resolve).toHaveBeenCalled();
+                });
+            });
+            it("rejects if the query isn't a string", () => {
+                searchParameters = {
+                    search: {
+                        query: "This won't work"
+                    }
+                };
+
+                return instance.search(searchParameters).then(jasmine.fail, () => {
+                    expect(bluebird.reject).toHaveBeenCalled();
+                });
+            });
+        });
+        describe(".validateSearch()", () => {
+            var searchParameters;
+
+            beforeEach(() => {
+                responseHandlerMock.resolveLinkHeaders.andReturn([]);
+            });
+            it("resolves the promise if the query list is falsy", () => {
+                spyOn(bluebird, "resolve").andCallThrough();
+
+                return instance.search().then(() => {
+                    expect(bluebird.resolve).toHaveBeenCalled();
+                });
+            });
+            it("calls bluebird.map on a single query", () => {
+                spyOn(bluebird, "map").andReturn(bluebird.resolve());
```

Comment:
Removed. We discussed the second part; I'm leaving the nested describes in in this case.